src/libraries/System.Net.Sockets/tests/FunctionalTests/ReceiveFrom.cs

```diff
@@ -154,6 +154,52 @@ public async Task ReceiveSent_UDP_Success(bool ipv4)
             }
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void ReceiveSent_SocketAddress_Success(bool ipv4)
+        {
+            //const int Offset = 10;
+            const int DatagramSize = 256;
+            const int DatagramsToSend = 16;
+
+            IPAddress address = ipv4 ? IPAddress.Loopback : IPAddress.IPv6Loopback;
+            using Socket server = new Socket(address.AddressFamily, SocketType.Dgram, ProtocolType.Udp);
+            using Socket client = new Socket(address.AddressFamily, SocketType.Dgram, ProtocolType.Udp);
+
+            client.BindToAnonymousPort(address);
+            server.BindToAnonymousPort(address);
+
+            byte[] sendBuffer = new byte[DatagramSize];
+            byte[] receiveBuffer = new byte[DatagramSize];
+
+            SocketAddress serverSA = server.LocalEndPoint.Serialize();
+            SocketAddress clientSA = client.LocalEndPoint.Serialize();
+            SocketAddress sa = new SocketAddress(address.AddressFamily);
+
+            Random rnd = new Random(0);
+
+            for (int i = 0; i < DatagramsToSend; i++)
+            {
+                rnd.NextBytes(sendBuffer);
+                client.SendTo(sendBuffer.AsSpan(), SocketFlags.None, serverSA);
+
+                int readBytes = server.ReceiveFrom(receiveBuffer, SocketFlags.None, sa);
+                Assert.Equal(sa, clientSA);
+                Assert.Equal(client.LocalEndPoint, client.LocalEndPoint.Create(sa));
+                Assert.True(new Span<byte>(receiveBuffer, 0, readBytes).SequenceEqual(sendBuffer));
+
+                // and send it back to make sure it works.
+                rnd.NextBytes(sendBuffer);
+                server.SendTo(sendBuffer, SocketFlags.None, sa);
+                readBytes = client.ReceiveFrom(receiveBuffer, SocketFlags.None, sa);
+                Assert.Equal(sa, serverSA);
+                Assert.Equal(server.LocalEndPoint, server.LocalEndPoint.Create(sa));
+                Assert.True(new Span<byte>(receiveBuffer, 0, readBytes).SequenceEqual(sendBuffer));
+
+            }
+        }
```

Comment:
This test fully covers everything added? Presumably we need tests for the new SocketAddress surface area?

Reply:
I will add more tests for SocketAddress. While this should cover the main use I think I can add some more test for invalid cases.

Reply:
I added more tests to cover more including short SocketAddress passed in and new SocketAddress API.